main.go

```diff
@@ -0,0 +1,47 @@
+// tail | (c) 2020 NETWAYS GmbH | GPLv2+
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+)
+
+func main() {
+	readAndPrintFile()
+}
+
+func readAndPrintFile() {
+	buf := bufio.NewReader(os.Stdin)
+	var lineCounter int
+	var allData [][]byte
+
+	for {
+		data, dataErr := buf.ReadBytes('\n')
+		if dataErr != nil && dataErr != io.EOF {
+			fmt.Fprintln(os.Stderr, dataErr)
+			os.Exit(1)
+		}
+		lineCounter++
+		allData = append(allData, data)
+		if dataErr == io.EOF {
+			if bytes.Compare(data, []byte("")) == 0 {
+				lineCounter--
+			}
+			break
+		}
+	}
+
+	if lineCounter < 10 {
+		lineCounter = 0
+	} else {
+		lineCounter = lineCounter - 10
+	}
+
+	for i := lineCounter; i < len(allData); i++ {
+		fmt.Printf("%s", allData[i])
```

Comment:
How did you output byte arrays like this in previous projects?

Reply:
os.Stdout.Write?

Reply:
Exactly!